tests/tacacs/test_authorization.py

```diff
@@ -0,0 +1,331 @@
+import crypt
+import paramiko
+import pytest
+
+from .test_ro_user import ssh_remote_run
+from tests.common.helpers.assertions import pytest_assert
+from .utils import stop_tacacs_server, start_tacacs_server
+
+logger = logging.getLogger(__name__)
+
+TIMEOUT_LIMIT   = 120
+
+def ssh_connect_remote(remote_ip, remote_username, remote_password):
+    ssh = paramiko.SSHClient()
+    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh.connect(remote_ip, username=remote_username, password=remote_password, allow_agent=False, look_for_keys=False, auth_timeout=TIMEOUT_LIMIT)
+    return ssh
+    
+
+def check_ssh_connect_remote_failed(remote_ip, remote_username, remote_password):
+    login_failed = False
+    try:
+        ssh_client_local = ssh_connect_remote(remote_ip, remote_username, remote_password)
+    except paramiko.ssh_exception.AuthenticationException as e:
+        login_failed = True
+    
+    pytest_assert(login_failed == True)
+
+def ssh_run_command(ssh_client, command):
+    stdin, stdout, stderr = ssh_client.exec_command(command, timeout=TIMEOUT_LIMIT)
+    exit_code = stdout.channel.recv_exit_status()
+    stdout_lines = stdout.readlines()
+    stderr_lines = stderr.readlines()
+    return exit_code, stdout_lines, stderr_lines
+
+def check_ssh_output(res, exp_val):
+    content_exist = False
+    for l in res:
+        if exp_val in l:
+            content_exist = True
+            break
+    pytest_assert(content_exist)
+
+@pytest.fixture
+def remote_user_client(duthosts, enum_rand_one_per_hwsku_hostname, creds_all_duts):
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    dutip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
+    ssh_client = ssh_connect_remote(dutip, creds_all_duts[duthost]['tacacs_authorization_user'],
+                         creds_all_duts[duthost]['tacacs_authorization_user_passwd'])
+    yield ssh_client
+    ssh_client.close()
+
+@pytest.fixture
+def local_user_client(duthosts, enum_rand_one_per_hwsku_hostname, creds_all_duts):
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    dutip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
+    ssh_client = paramiko.SSHClient()
+    ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    yield ssh_client
+    ssh_client.close()
+
+def test_authorization_tacacs_only(localhost, duthosts, enum_rand_one_per_hwsku_hostname, creds_all_duts, check_tacacs, remote_user_client):
```

Comment:
Please skip the test cases in this file for image built on 202106 and previous branches. The feature is not available there.